### PR TITLE
groups: Add loading state for notifications view

### LINKED ISSUE #1941: groups: notifications screen should show a loading state

### PR DESCRIPTION
**Pull request by patosullivan**
Fixes #1941 

Ended up just adding a flag to hark state for this. Also renamed the markRead request state stuff to avoid any future confusion.

https://user-images.githubusercontent.com/1221094/220432310-0e25a0e0-fe28-4bd1-85bb-9e5c18c4cd9e.mov

### CODE REVIEW

File: ui/src/notifications/useNotifications.tsx [hunk 1]
```diff
@@ -67,6 +68,14 @@ export const isComment = (yarn: Yarn) =>
...
   yarn.con.some((con) => con === ' replied to your message “');
 
+const startHark = async () => {
+  await useHarkState.getState().start();
+};
+
+const retrieveHark = async () => {
+  await useHarkState.getState().retrieve();
+};
+
```
Comment (arthyn) [inline ui/src/notifications/useNotifications.tsx:78]: these seem unused

**Review by arthyn — APPROVED** [review]
lgtm, small nit

**Review by jamesacklin — COMMENTED** [review]
Looks great. Gonna PR this PR with some additions to the Group Home screen so we don't have a completely empty state. Stay tuned.

File: ui/src/notifications/useNotifications.tsx (hunk 1 above)
Comment (patosullivan) [reply ui/src/notifications/useNotifications.tsx:78]: Oh, yeah, I need to remove those.

**Review by jamesacklin — CHANGES_REQUESTED** [review]
A freshly-booted ship with no notifications shows the skeleton loader forever: 

image

As does the Group Home of a completely empty group:
image

**Comment by patosullivan** [comment]
> A freshly-booted ship with no notifications shows the skeleton loader forever:
> 
> image
> 
> As does the Group Home of a completely empty group: image

Interesting, I tried the second case and didn't see that. Will investiage.

**Comment by patosullivan** [comment]
@jamesacklin now looks like this:

image
image

**Review by jamesacklin — APPROVED** [review]

File: ui/src/notifications/Notifications.tsx [hunk 2]
```diff
@@ -161,20 +188,34 @@ export default function Notifications({
...
-              </li>
-            ))}
-          </ul>
-        </div>
-      ))}
+      {loaded ? (
+        notifications.length === 0 ? (
+          <div className="flex w-full items-center justify-center">
+            <span className="text-xl font-semibold">
+              No notifications for {group ? 'this' : 'any'} group.
```
Comment (jamesacklin) [inline ui/src/notifications/Notifications.tsx:195]: Perfect, we'll fold in the root-level first-run wayfinding messages here. We can also do something group-specific.

cc @lukestiles

**Pull request merged**